app/Http/Controllers/GenreController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Genre;
+use Exception;
+use Illuminate\Http\Request;
+
+class GenreController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(Request $request)
+    {
+        try {
+            $perPage = $request->get('per_page', 10);
+            $genres = Genre::paginate($perPage);
+
+            return response()->json($genres, 200);
+        } catch (Exception $e) {
+            return response()->json([
+                'message' => 'An error occurred while retrieving genres.',
+            ], 400);
+        }
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        try {
+            $validated = $request->validate([
+                'name' => 'required|string|max:255|unique:genres',
+            ]);
+            $genre = Genre::create([
+                'name' => $validated['name'],
+            ]);
+
+            return response()->json($genre, 201);
+        } catch (Exception $e) {
+            return response()->json([
```

Comment:
@damir-tahirovic If you do not specify the status code, the default is 200. Maybe it's best to get the status code directly from the exception you are catching.